src/main/proto/upayload.proto

```diff
@@ -38,16 +38,17 @@
 // the pointer address to the data. If data is passed by value, the data is stored in the
 // value field. The length field is required if data is passed by reference. The format
 message UPayload {
-    oneof data {
-        fixed64 reference = 1; // Data is passed by reference so this is the pointer address
-        bytes value = 2;        // Data is passed by value so value contains a copy of the data
-    }
-
+    // Data is passed by value so value contains a copy of the data
+    bytes value = 1;       
+    
     // Length of the data in bytes, required if data is passed by reference only
-    optional int32 length = 3;
+    optional int32 length = 2;
 
     // Serialization format of the data
-    UPayloadFormat format = 4;
+    UPayloadFormat format = 3;
+
+    //Payload type
+    UPayloadType type = 4;
```

Comment:
hum why don't you like the oneof? with this you can determine what is populated. Regardless I want to wait for the zenoh discussions to conclude before we change this proto.